src/videojs-hls.js

```diff
@@ -377,12 +384,7 @@ videojs.Hls.prototype.cancelSegmentXhr = function() {
  * Abort all outstanding work and cleanup.
  */
 videojs.Hls.prototype.dispose = function() {
-  var player = this.player();
-
-  // remove event handlers
-  player.off('timeupdate', this.fillBuffer);
-  player.off('timeupdate', this.drainBuffer);
-  player.off('waiting', this.drainBuffer);
+  this.stopCheckingBuffer_();
```

Comment:
We could try to be more clever about when buffer checking was in effect but trying to be clever *was* the reason for these issues in the first place. This iteration does some extra work when no video is loaded or buffering has completed but it makes the tech lifecycle more consistent. The extra checks don't even show up in the profiler for me so I feel like this is reasonable.